.github/workflows/build_and_test_upstream_ci_image.yml

```diff
@@ -0,0 +1,31 @@
+name: "Build and test upstream CI image"
+
+# Controls when the action will run. 
```

Comment:
nit: I'm not sure how much of this 'boilerplate documentation we need.  If anything, I think the format of the github actions workflows should be documented in one place to make the yaml files a bit easier to read.  I'm open to suggestions.  If you think it should be kept here that's fine too.

Reply:
Inline documentation always help. 
I love the idea of adding github action workflows documentation. 
I will try to add separate document about github actions not as a part of this PR .